scipy/meson.build

```diff
@@ -6,6 +6,7 @@ cython_c_args = []
 if is_windows
   # For mingw-w64, link statically against the UCRT.
   gcc_link_args = ['-lucrt', '-static']
+  add_global_arguments('-fallow-argument-mismatch', language: 'fortran')
```

Comment:
This still didn't fix the scalar-rank warning:
```
../scipy/sparse/linalg/_eigen/arpack/ARPACK/SRC/cnaup2.f:503:35:

  392 |             call ivout (logfil, 1, iter, ndigit,
      |                                   2
......
  503 |             call ivout (logfil, 3, kp, ndigit,
      |                                   1
Warning: Rank mismatch between actual argument at (1) and actual argument at (2) (scalar and rank-1)
```

Reply:
I believe `-fallow-argument-mismatch` prevents a build failure on newer compilers (see https://github.com/scipy/scipy/pull/11842), but doesn't stop the warning. The warning needs an actual fix of the code.